credentials/src/main/java/me/uport/sdk/credentials/Credentials.kt

```diff
@@ -151,9 +154,113 @@ class Credentials(
         payload["claim"] = claim
         payload["vc"] = verifiedClaims ?: emptyList<String>()
         payload["callback"] = callbackUrl ?: ""
+
         return this.signJWT(payload, expiresInSeconds ?: 600L)
     }
 
+    /**
+     * Creates a W3C compliant [verifiable credential](https://w3c.github.io/vc-data-model/#credentials)
+     * serialized as a JWT.
+     *
+     * @param subject the subject of the claim in the [credential]
+     *      This becomes the `sub` field in the JWT
+     * @param credential the credential details; contains the claim being made about the [subject]
+     *      This becomes the `vc` field in the JWT
+     * @param notValidBefore [**optional**] the UNIX timestamp that marks the beginning of the validity period.
+     *      If this is not specified, the current timestamp is used as given by [clock]
+     *      This becomes the `nbf` field of the JWT. It is measured in seconds.
+     * @param validityPeriod [**optional**] the number of seconds of validity of this credential.
+     *      This influences the `exp` field in the JWT
+     *      If this is negative, there will be no expiry date set on the credential.
+     * @param audience [**optional**] the intended audience of this credential.
+     *      This results in the `aud` field of the JWT. If it is `null`, no audience will be set for the JWT.
+     * @param id [**optional**] the ID of this credential.
+     *      This becomes the `jti` field in the resulting JWT.
+     */
+    suspend fun createVerifiableCredential(
+        subject: String,
+        credential: CredentialParams,
+        notValidBefore: Long = clock.nowMs() / 1000L,
+        validityPeriod: Long = -1L,
+        audience: String? = null,
+        id: String? = null
+    ): String {
+
+        val payload = mutableMapOf<String, Any>()
+        payload["sub"] = subject
+
+        //add defaults if they are not set
+        val processedCredential = credential.copy(
+            context = credential.context.toMutableSet().apply { add("https://www.w3.org/2018/credentials/v1") }.toList(),
+            type = credential.type.toMutableSet().apply { add("VerifiableCredential") }.toList()
+        )
+
+        payload["vc"] = processedCredential
+        payload["nbf"] = notValidBefore
+        payload["iat"] = notValidBefore //for backward compatibility
+        if (validityPeriod >= 0) {
+            val exp = notValidBefore + validityPeriod
+            payload["exp"] = exp
+        }
+        if (audience != null) {
+            payload["aud"] = audience
+        }
+        if (id != null) {
+            payload["jti"] = id
+        }
+
+        return this.signJWT(
```

Comment:
`did-jwt-vc` provides pure functions for generating the JWTs, and no longer manages the state of the signer as `uport-credentials` did.  It seemed reasonable to me to keep the responsibility narrowly focused to enforcing the data structures from the specs, and passing in a signer that gets created elsewhere.  There will also be a library for creating the uport/serto-specific messages, and potentially other things, that need to use the signer as well.  Do you think these kinds of patterns are something that we should standardize between the libs for different languages?

Reply:
The signer is externally managed for this library as well.
It gets passed as a param to the `Credentials` constructor.

Perhaps I misunderstood the question here. Please rephrase if I did not answer it as expected.